File: classes/parte-8-como-funcionan-los-lenguajes/130-gestion-manual-de-memoria-c-malloc-free/implementaciones/c/main.c
#include <stdio.h>
#include <stdlib.h>

int main(void) {
    long n;
    if (scanf("%ld", &n) != 1) return 1;
    long *arr = malloc(n * sizeof(long));
    long suma = 0;
    for (long i = 0; i < n; i++) {
        arr[i] = i + 1;
        suma += arr[i];
    }
    printf("reservado=%ld suma=%ld\n", n, suma);
    free(arr);
    return 0;
}

File: classes/parte-8-como-funcionan-los-lenguajes/130-gestion-manual-de-memoria-c-malloc-free/implementaciones/go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	line, _ := bufio.NewReader(os.Stdin).ReadString('\n')
	n, _ := strconv.Atoi(strings.TrimSpace(line))
	arr := make([]int, n)
	suma := 0
	for i := 0; i < n; i++ {
		arr[i] = i + 1
		suma += arr[i]
	}
	fmt.Printf("reservado=%d suma=%d\n", n, suma)
}
